File: sdk/python/examples/49_include_contents.py
# Copyright (c) 2025 Agentspan
# Licensed under the MIT License. See LICENSE file in the project root for details.

"""Include Contents — control context passed to sub-agents.

When ``include_contents="none"``, a sub-agent starts with a clean slate
and does NOT see the parent agent's conversation history. This is useful
for sub-agents that should work independently without being influenced
by prior messages.

Requirements:
    - Conductor server with include_contents support
    - AGENTSPAN_SERVER_URL=http://localhost:8080/api in .env or environment
    - AGENT_LLM_MODEL=openai/gpt-4o-mini in .env or environment
"""

from agentspan.agents import Agent, AgentRuntime, tool
from settings import settings


@tool
def summarize_text(text: str) -> dict:
    """Summarize a piece of text.

    Args:
        text: The text to summarize.

    Returns:
        Dictionary with the summary.
    """
    words = text.split()
    return {"summary": " ".join(words[:20]) + "...", "word_count": len(words)}


# This sub-agent won't see the parent's conversation history
independent_summarizer = Agent(
    name="independent_summarizer_49",
    model=settings.llm_model,
    instructions="You are a summarizer. Summarize any text given to you concisely.",
    tools=[summarize_text],
    include_contents="none",  # No parent context
)

# This sub-agent WILL see the parent's conversation history (default)
context_aware_helper = Agent(
    name="context_aware_helper_49",
    model=settings.llm_model,
    instructions="You are a helpful assistant that builds on prior conversation context.",
)

coordinator = Agent(
    name="coordinator_49",
    model=settings.llm_model,
    instructions=(
        "You coordinate tasks. Route summarization requests to "
        "independent_summarizer_49 and general questions to context_aware_helper_49."
    ),
    agents=[independent_summarizer, context_aware_helper],
    strategy="handoff",
)

with AgentRuntime() as runtime:
    result = runtime.run(
        coordinator,
        "Please summarize this: 'The quick brown fox jumps over the lazy dog. "
        "This sentence contains every letter of the alphabet and is commonly "
        "used for typography testing.'",
    )
    result.print_result()

File: sdk/typescript/examples/49-include-contents.ts
/**
 * 49 - Include Contents — control context passed to sub-agents.
 *
 * When `includeContents: 'none'`, a sub-agent starts with a clean slate
 * and does NOT see the parent agent's conversation history.
 *
 * Requirements:
 *   - Conductor server with include_contents support
 *   - AGENTSPAN_SERVER_URL=http://localhost:6767/api as environment variable
 *   - AGENTSPAN_LLM_MODEL=openai/gpt-4o-mini as environment variable
 */

import { z } from 'zod';
import { Agent, AgentRuntime, tool } from '../src/index.js';
import { llmModel } from './settings.js';

// -- Tool --------------------------------------------------------------------

const summarizeText = tool(
  async (args: { text: string }) => {
    const words = args.text.split(/\s+/);
    return { summary: words.slice(0, 20).join(' ') + '...', word_count: words.length };
  },
  {
    name: 'summarize_text',
    description: 'Summarize a piece of text.',
    inputSchema: z.object({
      text: z.string().describe('The text to summarize'),
    }),
  },
);

// -- Agents ------------------------------------------------------------------

// This sub-agent won't see the parent's conversation history
export const independentSummarizer = new Agent({
  name: 'independent_summarizer_49',
  model: llmModel,
  instructions: 'You are a summarizer. Summarize any text given to you concisely.',
  tools: [summarizeText],
  includeContents: 'none', // No parent context
});

// This sub-agent WILL see the parent's conversation history (default)
export const contextAwareHelper = new Agent({
  name: 'context_aware_helper_49',
  model: llmModel,
  instructions: 'You are a helpful assistant that builds on prior conversation context.',
});

export const coordinator = new Agent({
  name: 'coordinator_49',
  model: llmModel,
  instructions:
    'You coordinate tasks. Route summarization requests to ' +
    'independent_summarizer_49 and general questions to context_aware_helper_49.',
  agents: [independentSummarizer, contextAwareHelper],
  strategy: 'handoff',
});

// -- Run ---------------------------------------------------------------------

// Only run when executed directly (not when imported for discovery)
async function main() {
  const runtime = new AgentRuntime();
  try {
    // Deploy to server. CLI alternative (recommended for CI/CD):
    //   agentspan deploy <module>
    // await runtime.deploy(coordinator);
    // await runtime.serve(coordinator);
    // Direct run for local development:
    // const runtime = new AgentRuntime();
    // try {
    const result = await runtime.run(
    coordinator,
    "Please summarize this: 'The quick brown fox jumps over the lazy dog. " +
    "This sentence contains every letter of the alphabet and is commonly " +
    "used for typography testing.'",
    );
    result.printResult();
  } finally {
    await runtime.shutdown();
    // }
}

if (process.argv[1]?.endsWith('49-include-contents.ts') || process.argv[1]?.endsWith('49-include-contents.js')) {
  main().catch(console.error);
}
